tests/ssg_test_suite/common.py

```diff
@@ -233,29 +248,155 @@ def _make_file_root_owned(tarinfo):
+    # it on success. Wrap in a try/except block and reraise the original
+    # exception after removing the temporary directory.
+    try:
+        # Load product's YAML file if present. This will allow us to parse
+        # tests in the context of the product we're executing under.
+        product_yaml = dict()
+        if product:
+            yaml_path = product_yaml_path(SSG_ROOT, product)
+            product_yaml = load_product_yaml(yaml_path)
+
+        # Below we could run into a DocumentationNotComplete error. However,
+        # because the test suite isn't executed in the context of a particular
+        # build (though, ideally it would be linked), we may not know exactly
+        # whether the top-level rule/profile we're testing is actually
+        # completed. Thus, forcibly set the required property to bypass this
+        # error.
+        product_yaml['cmake_build_type'] = 'Debug'
+
+        # Note that we're not exactly copying 1-for-1 the contents of the
+        # directory structure into the temporary one. Instead we want a
+        # flattened mapping with all rules in a single top-level directory
+        # and all tests immediately contained within it. That is:
+        #
+        # /group_a/rule_a/tests/something.pass.sh -> /rule_a/something.pass.sh
+        for dirpath, dirnames, _ in walk_through_benchmark_dirs(product):
+            # Skip anything that isn't obviously a rule.
+            if "tests" not in dirnames or not is_rule_dir(dirpath):
+                continue
+
+            # Load rule content in our environment. We use this to satisfy
+            # some implied properties that might be used in the test suite.
+            rule_path = get_rule_dir_yaml(dirpath)
+            rule = RuleYAML.from_yaml(rule_path, product_yaml)
+
+            # Note that most places would check prodtype, but we don't care
+            # about that here: if the rule is available to the product, we
+            # load and parse it anyways as we have no knowledge of the
+            # top-level profile or rule passed into the test suite.
+            prodtypes = parse_prodtype(rule.prodtype)
+
+            # Our local copy of env_yaml needs some properties from rule.yml
+            # for completeness.
+            local_env_yaml = dict()
+            local_env_yaml.update(product_yaml)
+            local_env_yaml['rule_id'] = rule.id_
+            local_env_yaml['rule_title'] = rule.title
+            local_env_yaml['products'] = prodtypes
+
+            # Create the destination directory.
+            dest_path = os.path.join(tmpdir, rule.id_)
+            os.mkdir(dest_path)
+
+            # Walk the test directory, writing all tests into the output
+            # directory, recursively.
+            tests_dir_path = os.path.join(dirpath, "tests")
+            tests_dir_path = os.path.abspath(tests_dir_path)
+            for dirpath, dirnames, filenames in os.walk(tests_dir_path):
+                for dirname in dirnames:
```

Comment:
Do we expect subdirectories in the `tests` directory? I couldn't find any.

Reply:
On my checkout I see tons of `tests/ocp4/e2e.yml` subdirectories:

```
./linux_os/guide/system/auditing/auditd_configure_rules/audit_privileged_commands/audit_rules_privileged_commands_sudo/tests/ocp4/e2e.yml
./linux_os/guide/system/auditing/package_audit_installed/tests/ocp4/e2e.yml
./linux_os/guide/system/logging/log_rotation/ensure_logrotate_activated/tests/ocp4/e2e.yml
./linux_os/guide/system/selinux/selinux_policytype/tests/ocp4/e2e.yml
```

(total of 393 results). 

I have no idea if/how they're used but my thought was we might as well jinja enable them while we're here.